### ISSUE TITLE
Subpage detection different to MediaWiki core

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticBreadcrumbLinks#23]
### Setup and configuration

- MediaWiki 	1.28.0 (fec3e69) 18:52, 1 March 2017
- Semantic MediaWiki	2.5.0 (a41d159) 08:42, 23 March 2017
- Semantic Breadcrumb Links	1.3.0 (be3f51c) 13:22, 2 January 2017
- PHP 	5.6.30-0+deb8u1 (apache2handler)
- MariaDB 	10.0.30-MariaDB-1~jessie

### Issue

When adding a page called [Page_/not_subpage](https://sandbox.semantic-mediawiki.org/wiki/Page_/not_subpage), not the blanc before the slash when MediaWiki does not see `not_subpage` as a subpage of [`Page_`](https://sandbox.semantic-mediawiki.org/w/index.php?title=Page&action=info) whereas SBL does. I think it will be better if SBL also accounts for the blanc.

Steps to reproduce the observation (recommendation is to use the [sandbox](http://sandbox.semantic-mediawiki.org)):
- See the links in the issue description.

**Comment by mwjames** [comment]
Expected to be fixed with #24.

**Comment by kghbln** [comment]
@mwjames Any objections on getting this and [that](https://github.com/SemanticMediaWiki/SemanticBreadcrumbLinks/compare/1.3.0...master) out to the wild as 1.3.1 after I verified the fix on sandbox.

**Comment by mwjames** [comment]
> Any objections

No.

**Comment by kghbln** [comment]
> No.

Cool, will do no Monday.

**Comment by kghbln** [comment]
@mwjames I think there is still an issue with the display of the title. The title should now be ["Page_/not subpage"](https://sandbox.semantic-mediawiki.org/wiki/Page_/not_subpage) instead of just "not subpage". The crumbs as such are fine now since they do not show.

**Comment by mwjames** [comment]
> The title should now be "Page_/not subpage" instead of just "not subpage".

I might be wrong but what about the `$wgNamespacesWithSubpages[NS_MAIN] = true;` setting?

**Comment by mwjames** [comment]
MW's own `Title::isSubpage` believes (`strpos( $this->getText(), '/' )` that "Page_/not subpage" is a subpage.

```
	public function isSubpage() {
		return MWNamespace::hasSubpages( $this->mNamespace )
			? strpos( $this->getText(), '/' ) !== false
			: false;
	}
```

**Comment by kghbln** [comment]
Yes, this is already set:
```php
$wgNamespacesWithSubpages = array_fill(
      0, 4000, true
      );
```

**Comment by kghbln** [comment]
> MW's own Title::isSubpage believes (strpos( $this->getText(), '/' ) that "Page_/not subpage" is a subpage.

This is a bit strange since [action `info`](https://sandbox.semantic-mediawiki.org/w/index.php?title=Page&action=info) should reveal this. Morever when I detected the issue yesterday the title was indeed showing "Page_/not subpage" as expected plus the breadcrumbs.

**Comment by mwjames** [comment]
I'll have a look at it.

**Comment by mwjames** [comment]
> Any objections on getting this and that out to the wild as 1.3.1 after

@kghbln As for the release process, now that the master contains the `~3.0` constraint and we don't want to create extra branches for the sake of a release, I'd propose the following workflow so that only the master is referencing to SMW `~3.0`.

- Before the release, remove `|~3.0` from `composer.json` so that the 1.3.1 can be released without a reference to `~3.0`
- Do the release process including tag creation
- After the tag has been created reapply `|~3.0` to `composer.json` so that the master can continue to work with both SMW 2.x and 3.x branches

We don't know what changes or breakages will happen in SMW 3.x but to avoid having a single release (e.g SBL 1.3.1) containing a constraint that may allow a reference to a release that contains breakages, the workflow above is expected to avoid that conflict.

**Comment by mwjames** [comment]
> I'll have a look at it.

Should be fixed with #26.

**Comment by kghbln** [comment]
> Should be fixed with #26.

Affirmative!
